### PR TITLE
Initial version of the job manager micro-service

### PR DESCRIPTION
**Pull request by nirmal070125**
Please note that the code under `modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/` is auto-generated via swagger2msf4j. 

org.wso2.carbon.das.jobmanager.core.impl.WorkersApiServiceImpl.java has the service implementation.

### CODE REVIEW

File: modules/components/org.wso2.carbon.das.jobmanager.core/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+This server was generated by the [swagger-codegen](https://github.com/swagger-api/swagger-codegen) project. By using the 
+[OpenAPI-Spec](https://github.com/swagger-api/swagger-core/wiki) from a remote server, you can easily generate a server stub.  This
+is an example of building a swagger-enabled JAX-RS server.
+
+This example uses the [JAX-RS](https://jax-rs-spec.java.net/) framework.
+
+To run the server, please execute the following:
+
+```
+mvn clean package jetty:run
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/README.md]: Are we implementing a JAX-RS based service or MSF4J?

File: modules/components/org.wso2.carbon.das.jobmanager.core/pom.xml [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+    <artifactId>org.wso2.carbon.das.jobmanager.core</artifactId>
+    <packaging>bundle</packaging>
+
+    <name>WSO2 Data Analytics Server - Job Manager</name>
+    <url>http://wso2.org</url>
+
+    <dependencies>
+		<dependency>
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/pom.xml]: indentation

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package org.wso2.carbon.das.jobmanager.core;
+
+import java.io.IOException;
+
+import javax.servlet.*;
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java]: wildcard imports

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
+package org.wso2.carbon.das.jobmanager.core;
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java]: licence

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiResponseMessage.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
+package org.wso2.carbon.das.jobmanager.core;
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiResponseMessage.java]: licence

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/WorkersApi.java [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+import org.wso2.carbon.das.jobmanager.core.dto.ExecutionPlanDTO;
+import org.wso2.carbon.das.jobmanager.core.dto.ExecutionPlanListDTO;
+import org.wso2.carbon.das.jobmanager.core.dto.WorkerDTO;
+import org.wso2.carbon.das.jobmanager.core.dto.WorkerListDTO;
+
+import org.wso2.msf4j.Microservice;
+import org.osgi.service.component.annotations.Component;
+
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/WorkersApi.java]: wildcard imports

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiException.java [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package org.wso2.carbon.das.jobmanager.core;
```
Comment (tishan89) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiException.java]: licence

File: modules/components/org.wso2.carbon.das.jobmanager.core/README.md (hunk 1 above)
Comment (nirmal070125) [inline modules/components/org.wso2.carbon.das.jobmanager.core/README.md]: @tishan89 All these are auto-generated files, will clean them up a bit. Thanks.

Comment (nirmal070125) [inline modules/components/org.wso2.carbon.das.jobmanager.core/README.md]: This was an auto-generated README. Need to remove it. thanks @tishan89

File: modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java (hunk 3 above)
Comment (nirmal070125) [inline modules/components/org.wso2.carbon.das.jobmanager.core/src/gen/java/org/wso2/carbon/das/jobmanager/core/ApiOriginFilter.java]: These were auto-generated code. Can fix it now but if we had to regenerate the code, we have the same issues.

**Pull request merged**